core/src/main/java/net/adoptopenjdk/icedteaweb/resources/downloader/BaseResourceDownloader.java

```diff
@@ -152,7 +154,20 @@ private Resource tryDownloading(final URL downloadFrom) throws IOException {
     private CloseableConnection getDownloadConnection(final URL location) throws IOException {
         final Map<String, String> requestProperties = new HashMap<>();
         requestProperties.put(ACCEPT_ENCODING_HEADER, PACK_200_OR_GZIP);
-        return ConnectionFactory.openConnection(location, HttpMethod.GET, requestProperties);
+        return ConnectionFactory.openConnection(location, HttpMethod.GET, requestProperties, getTimeoutValue(ConfigurationConstants.KEY_HTTPCONNECTION_CONNECT_TIMEOUT), getTimeoutValue(ConfigurationConstants.KEY_HTTPCONNECTION_READ_TIMEOUT));
+    }
+
+    private int getTimeoutValue(final String key) {
+        int timeout = 0;
+        final String value = JNLPRuntime.getConfiguration().getProperty(key);
+        if (value != null && value.trim().length() != 0) {
+            try {
+                timeout = Integer.valueOf(value);
+            } catch (NumberFormatException e) {
+                LOG.error(IcedTeaWebConstants.DEFAULT_ERROR_MESSAGE, e);
```

Comment:
Please use something else than `DEFAULT_ERROR_MESSAGE`, maybe
`Log.error("Could not pares {} with value {} - reason {}", key, value, e.getMessage());`

Reply:
done